### PR TITLE
Rafactoring + more tests

### PR DESCRIPTION
**Pull request by catena2w**

### CODE REVIEW

File: src/main/scala/com/circlegain/exchanges/binance/rest/BinancePrivateRestClient.scala [hunk 1]
```diff
@@ -117,49 +116,58 @@ final class BinancePrivateRestClient(apiKey: String, secretKey: String)(
...
-    accountService.getAccountInfo.getWallet.getBalances.asScala.values.toSeq.map(b => convert(b))
+    accountService
+      .getAccountInfo
+      .getWallet
+      .getBalances
+      .asScala
+      .values
+      .toSeq
+      .map(b => convert(b))
+      .filter(_.totalBalance > 0)
```
Comment (eanea) [inline src/main/scala/com/circlegain/exchanges/binance/rest/BinancePrivateRestClient.scala:137]: add showZeroBalances

**Review by eanea — CHANGES_REQUESTED** [review]

File: src/main/scala/com/circlegain/exchanges/binance/rest/BinancePrivateRestClient.scala (hunk 1 above)
Comment (catena2w) [reply src/main/scala/com/circlegain/exchanges/binance/rest/BinancePrivateRestClient.scala]: fixed

File: src/main/scala/com/circlegain/exchanges/binance/rest/BinancePrivateRestClient.scala [hunk 2]
```diff
@@ -117,49 +117,60 @@ final class BinancePrivateRestClient(apiKey: String, secretKey: String)(
...
    */
-  override def getBalances: Task[Seq[Balance]] = Task.delay(
-    accountService.getAccountInfo.getWallet.getBalances.asScala.values.toSeq.map(b => convert(b))
+  override def getBalances(showZeroBalances: Boolean = false): Task[Seq[Balance]] = Task.delay(
+    accountService
+      .getAccountInfo
+      .getWallet
+      .getBalances
+      .asScala
+      .filter(b => showZeroBalances || BigDecimal(b._2.getTotal) > 0)
```
Comment (eanea) [inline src/main/scala/com/circlegain/exchanges/binance/rest/BinancePrivateRestClient.scala:135]: I think should be >= 0

File: src/main/scala/com/circlegain/exchanges/core/rest/AuthenticatedRestExchange.scala [hunk 3]
```diff
@@ -46,12 +47,28 @@ trait AuthenticatedRestExchange extends UnauthenticatedRestExchange {
...
+  def getBalances(showZeroBalances: Boolean = false): Task[Seq[Balance]]
+
+  /**
+   * Get balances of the token `token`
+   */
+  def getBalance(token: Token): Task[Balance] = {
+    getBalances(showZeroBalances = true).map { allBalances =>
+      val requestedBalance = allBalances.find(_.cur == token)
+      val ts               = allBalances.headOption.map(_.updated).getOrElse(NTP.correctedTime())
+      requestedBalance.getOrElse(Balance(ts, token, 0, 0))
```
Comment (eanea) [inline src/main/scala/com/circlegain/exchanges/core/rest/AuthenticatedRestExchange.scala:59]: It's incorrect. You return zero balance, even such balance doesn't exist on the server.

**Review by eanea — CHANGES_REQUESTED** [review]

**Pull request merged**